pyvela/pyvela/pyvela_script.py

```diff
@@ -128,34 +135,62 @@ def validate_input(args):
         args.truth
     ), f"Truth par file {args.truth} not found.  Make sure the path is correct."
 
+    if args.restart:
+        args.force_rewrite = True
     assert args.force_rewrite or not os.path.isdir(
         args.outdir
     ), f"The output directory {args.outdir} already exists! Use `-f` option to force overwrite."
 
 
 def prepare_outdir(args):
-    if args.force_rewrite and os.path.isdir(args.outdir):
+    if args.force_rewrite and os.path.isdir(args.outdir) and not args.restart:
         shutil.rmtree(args.outdir)
 
-    os.mkdir(args.outdir)
+    if not args.restart and not os.path.isdir(args.outdir):
+        os.mkdir(args.outdir)
 
-    shutil.copy(args.par_file, args.outdir)
+    if not os.path.exists(f"{args.outdir}/{os.path.basename(args.par_file)}"):
+        shutil.copy(args.par_file, args.outdir)
 
     if args.jlso_file is None:
-        shutil.copy(args.tim_file, args.outdir)
+        if not os.path.exists(f"{args.outdir}/{os.path.basename(args.tim_file)}"):
+            shutil.copy(args.tim_file, args.outdir)
     else:
-        shutil.copy(args.jlso_file, args.outdir)
+        if not os.path.exists(f"{args.outdir}/{os.path.basename(args.jlso_file)}"):
+            shutil.copy(args.jlso_file, args.outdir)
 
-    if args.prior_file is not None:
+    if args.prior_file is not None and not os.path.exists(
+        f"{args.outdir}/{os.path.basename(args.prior_file)}"
+    ):
         shutil.copy(args.prior_file, args.outdir)
 
-    if args.truth is not None:
+    if args.truth is not None and not os.path.exists(
+        f"{args.outdir}/{os.path.basename(args.truth)}"
+    ):
```

Comment:
Since we are ignoring input files when the resume option is given, these copy operations should not be executed in that case. That way wrong/invalid files won't get in the output directory.

Reply:
I don't know that this would happen, but I have changed it in any case.